### PR TITLE
New StringParse* functions

### PR DESCRIPTION
**Pull request by inashivb**
Add a new set of StringParse* functions and use them for strict parsing in all parser, configuration and setup functions.

### CODE REVIEW

File: src/util-byte.h [hunk 1]
```diff
@@ -253,6 +255,111 @@ int ByteExtractStringInt16(int16_t *res, int base, uint16_t len, const char *str
...
+ * \param len Number of bytes to extract (23 max or 0 for unbounded)
+ * \param str String to extract from
+ *
+ * \return n Number of bytes extracted on success
+ * \return -1 On error
+ */
+int StringParseInt32(int32_t *res, int base, uint16_t len, const char *str);
+
+/**
+ * Extract signed integer value from a string as uint16_t strictly.
```
Comment (jlucovsky) [inline src/util-byte.h]: nit: s/uint16_t/int16_t/

File: src/util-byte.h [hunk 2]
```diff
@@ -253,6 +255,111 @@ int ByteExtractStringInt16(int16_t *res, int base, uint16_t len, const char *str
...
+ * \param len Number of bytes to extract (23 max or 0 for unbounded)
+ * \param str String to extract from
+ *
+ * \return n Number of bytes extracted on success
+ * \return -1 On error
+ */
+int StringParseUint8(uint8_t *res, int base, uint16_t len, const char *str);
+
+/**
+ * Extract signed integer value from a string as uint64_t strictly.
```
Comment (jlucovsky) [inline src/util-byte.h]: nit: s/uint64_t/int64_t/

File: src/util-byte.h [hunk 3]
```diff
@@ -253,6 +255,111 @@ int ByteExtractStringInt16(int16_t *res, int base, uint16_t len, const char *str
...
+ * \param len Number of bytes to extract (23 max or 0 for unbounded)
+ * \param str String to extract from
+ *
+ * \return n Number of bytes extracted on success
+ * \return -1 On error
+ */
+int StringParseInt64(int64_t *res, int base, uint16_t len, const char *str);
+
+/**
+ * Extract signed integer value from a string as uint32_t strictly.
```
Comment (jlucovsky) [inline src/util-byte.h]: nit: s/uint32_t/int32_t/

File: src/util-byte.h [hunk 4]
```diff
@@ -253,6 +255,111 @@ int ByteExtractStringInt16(int16_t *res, int base, uint16_t len, const char *str
...
+ * \param len Number of bytes to extract (23 max or 0 for unbounded)
+ * \param str String to extract from
+ *
+ * \return n Number of bytes extracted on success
+ * \return -1 On error
+ */
+int StringParseInt16(int16_t *res, int base, uint16_t len, const char *str);
+
+/**
+ * Extract signed integer value from a string as uint8_t strictly.
```
Comment (jlucovsky) [inline src/util-byte.h]: nit: s/uint8_t/int8_t/

**Comment by inashivb** [comment]
Replaced with #4488

**Pull request closed without merging**